daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/parsers/PState.scala

```diff
@@ -166,6 +168,9 @@ final class PState private (
   private val discriminatorStack = MStackOfBoolean()
   discriminatorStack.push(false)
 
+  private val changedVariablesStack = new MStackOf[mutable.MutableList[GlobalQName]]()
```

Comment:
Can you add a comment describing what this is? It's not clear what a stack of lists of qnames is used for.